pkg/ruler/rules/objectclient/rule_store.go

```diff
@@ -172,6 +163,58 @@ func (o *RuleStore) DeleteNamespace(ctx context.Context, userID, namespace strin
 	return nil
 }
 
+func (o *RuleStore) loadRuleGroupsConcurrently(ctx context.Context, rgObjects []chunk.StorageObject) (map[string]rules.RuleGroupList, error) {
+	ch := make(chan string, len(rgObjects))
+
+	for _, obj := range rgObjects {
+		level.Debug(util.Logger).Log("msg", "listing rule group", "key", obj.Key)
```

Comment:
You may want to put this debug into fetching goroutine just before calling `o.getRuleGroup(gCtx, key)`, to avoid logging too early.